src/ClaimService.Business/Features/Categories/Commands/Edit/EditCategoryValidator.cs

```diff
@@ -0,0 +1,71 @@
+﻿using FluentValidation;
+using LT.DigitalOffice.ClaimService.Business.Shared.Enums;
+using LT.DigitalOffice.Kernel.Validators;
+using Microsoft.AspNetCore.JsonPatch;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ClaimService.Business.Features.Categories.Commands.Edit;
+
+public class EditCategoryValidator : BaseEditRequestValidator<EditCategoryRequest>, IEditCategoryValidator
+{
+  private async Task HandleInternalPropertyValidation(
+    Operation<EditCategoryRequest> requestedOperation,
+    ValidationContext<JsonPatchDocument<EditCategoryRequest>> context,
+    CancellationToken cancellationToken)
+  {
+    RequestedOperation = requestedOperation;
+    Context = context;
+
+    #region Paths
+
+    AddСorrectPaths(new()
+    {
+      nameof(EditCategoryRequest.Name),
+      nameof(EditCategoryRequest.Color)
+    });
+
+    AddСorrectOperations(nameof(EditCategoryRequest.Name), new() { OperationType.Replace });
+    AddСorrectOperations(nameof(EditCategoryRequest.Color), new() { OperationType.Replace });
+
+    #endregion
+
+    #region Name
+
+    AddFailureForPropertyIf(
+      nameof(EditCategoryRequest.Name),
+      x => x == OperationType.Replace,
+      new()
+      {
+        { x => !string.IsNullOrWhiteSpace(x.value.ToString().Trim()), "Name can't be null or empty." },
+        { x => x.value.ToString().Trim().Length < 21, "Name is too long." }
+      },
+      CascadeMode.Stop);
+
+    #endregion
+
+    #region Color
+
+    await AddFailureForPropertyIfAsync(
```

Comment:
use just AddFailureForPropertyIf(), then there will be no need in Task.FromResult()